doc/manual/parallel-computing.rst

```diff
@@ -929,6 +929,55 @@ implementation simply executes an ``exit()`` call on the specified remote worker
 
+Network requirements for LocalManager and SSHManager
+----------------------------------------------------
+Julia clusters are designed to be executed on already secured environments on infrastructure ranging from local laptops,
+to departmental clusters or even on the Cloud. This section covers network security requirements for the inbuilt ``LocalManager``
+and ``SSHManager``:
+
+- The master process does not listen on any port. It only connects out to the workers.
+
+- Each worker binds to only one of the local interfaces and listens on the first free port starting from 9009.
+
+- ``LocalManager``, i.e. ``addprocs(N)``, by default binds only to the loopback interface.
+  This means that workers consequently started on remote hosts, or anyone with malafide intentions
+  is unable to connect to the cluster. A ``addprocs(4)`` followed by a ``addprocs(["remote_host"])``
+  will fail. Some users may need to create a cluster comprising on their local system and a few remote systems. This can be done by
+  explicitly requesting ``LocalManager`` to bind to an external network interface via the ``restrict`` keyword
+  argument -  ``addprocs(4; restrict=false)``.
+
+- ``SSHManager``, i.e. ``addprocs(list_of_remote_hosts)`` launches workers on remote hosts via SSH.
+   It is to be noted that by default SSH is only used to launch Julia workers.
+   Subsequent, master-worker and worker-worker connections use plain, unencrypted TCP/IP sockets. The remote hosts
+   must have passwordless login enabled. Additional SSH flags or credentials may be specified via keyword
+   argument ``sshflags``.
+
+- ``addprocs(list_of_remote_hosts; tunnel=true, sshflags=<ssh keys and other flags>)`` is useful when we wish to use
+  SSH connections for master-worker too. A typical scenario for this is a local laptop running the Julia REPL(i.e., the master)
+  with the rest of the cluster on the Cloud, say on Amazon EC2. You will need to open only port 22 into the remote cluster, with
+  SSH clients authenticated via PKI. ``sshflags`` can specify ``-e <keyfile>`` for the same.
+
+  Note that worker-worker connections are still plain TCP and the local security policy on the remote cluster
+  must allow for free connections between worker nodes, at least for ports 9009 and above.
+
+  Securing and encrypting all worker-worker traffic (via SSH), or encrypting individual messages can be done via
+  a custom ClusterManager.
+
+Cluster cookie
+--------------
+All processes in a cluster share the same cookie which, by default, is a randomly generated string on the master process:
+
+- ``cluster_cookie()`` returns the cookie, ``cluster_cookie(cookie)`` sets it.
+- All connections are authenticated on both sides to ensure that only workers started by the master are allowed
+  to connect to each other.
+- The cookie must be passed to the workers at startup via argument ``--worker <cookie>``.
+  Custom ``ClusterManagers`` can retrieve the cookie on the master by calling
```

Comment:
be consistent about whether this is highlighted as code (the s probably should not be included if so)